ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/WidgetCard.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { ReactNode } from 'react';
+import { Skeleton, Card, Title } from '@patternfly/react-core';
+
+import { defaultChartHeight } from 'utils/chartUtils';
+import WidgetErrorEmptyState from './WidgetErrorEmptyState';
+
+type WidgetCardProps = {
+    title: string;
+    isLoading: boolean;
+    error: Error | null;
+    children: ReactNode;
+};
+
+const height = `${defaultChartHeight}px` as const;
+
+function WidgetCard({ title, isLoading, error, children }: WidgetCardProps) {
+    let cardContent: ReactNode;
+
+    if (isLoading && !error) {
+        cardContent = <Skeleton height={height} screenreaderText={`Loading ${title}`} />;
+    } else if (error) {
+        cardContent = (
+            <WidgetErrorEmptyState height={height} title="Unable to load data">
+                There was an error loading data for this widget
+            </WidgetErrorEmptyState>
+        );
+    } else {
+        cardContent = children;
+    }
+
+    return (
+        <Card>
+            <Title headingLevel="h2" className="pf-u-p-md">
+                {title}
+            </Title>
```

Comment:
I can see the value in having this wrapper for the loading/skeleton and errors states, but I don't think we should assume that every widget will only have a title. There could be subtitles, calls-to-action, etc. (I already know of a few places where we have already done this, or will do it when we migrate some of the *-Management sections to PF.

Let's leave it to the caller to set all of the "happy-path" content, including the header.

Reply:
Good idea, definitely easier to move the <Title> out and pass as a child element now instead of refactoring _n_ widgets later. I'll push this fix up.